### PR TITLE
Feature: Validator Registration

### PR DESCRIPTION
**Pull request by talekhinezh**
* Add Validator register/unregister
* Remove REv1!
* Update Scrypto Lib to latest `develop`
* Add Dynamic Validator Set integration tests

### CODE REVIEW

File: core-rust/core-api-server/src/core_api/conversions/addressing.rs [hunk 1]
```diff
@@ -336,9 +336,14 @@ fn to_mapped_substate_id(substate_id: SubstateId) -> Result<MappedSubstateId, Ma
                     EpochManagerOffset::EpochManager => {
                         (SubstateType::EpochManager, SubstateKeyType::EpochManager)
                     }
-                    EpochManagerOffset::ValidatorSet => {
-                        (SubstateType::EpochManager, SubstateKeyType::ValidatorSet)
-                    }
+                    EpochManagerOffset::CurrentValidatorSet => (
```
Comment (talekhinezh) [inline core-rust/core-api-server/src/core_api/conversions/addressing.rs:339]: "CurrentEpochSnapshot"

File: core-rust/core-api-server/src/core_api/conversions/addressing.rs [hunk 2]
```diff
@@ -336,9 +336,14 @@ fn to_mapped_substate_id(substate_id: SubstateId) -> Result<MappedSubstateId, Ma
...
+                        SubstateType::EpochManager,
+                        SubstateKeyType::CurrentValidatorSet,
+                    ),
+                    EpochManagerOffset::PreparingValidatorSet => (
```
Comment (talekhinezh) [inline core-rust/core-api-server/src/core_api/conversions/addressing.rs:343]: Change to:
"PotentialValidators"
"AspiringValidators"

File: core-rust/core-api-server/src/core_api/conversions/addressing.rs (hunk 1 above)
Comment (talekhinezh) [reply core-rust/core-api-server/src/core_api/conversions/addressing.rs:339]: "CurrentEpochValidators"

Comment (talekhinezh) [reply core-rust/core-api-server/src/core_api/conversions/addressing.rs:339]: "CurrentValidatorSet"

**Review by dhedey — APPROVED** [review]
Approving after in-person review - subject to substate naming updates, and sign-off from Theo/Les

**Comment by Theoklitos** [comment]
👍

File: core-rust/core-api-server/src/core_api/conversions/addressing.rs (hunk 2 above)
Comment (talekhinezh) [reply core-rust/core-api-server/src/core_api/conversions/addressing.rs:343]: This is on RadixEngine side so will update these names there in a separate PR.

**Pull request merged**